### PR TITLE
Graduate storageVersionHash field in the discovery doc to beta

### PR DESCRIPTION
**Pull request by caesarxuchao**
https://github.com/kubernetes/kubernetes/pull/72942 introduced this field as alpha. Now we graduate the field to beta according to https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/35-storage-version-hash.md.

/kind feature
/sig api-machinery

```release-note
The storageVersionHash feature is beta now. "StorageVersionHash" is a field in the discovery document of each resource. It allows clients to detect if the storage version of that resource has changed. Its value must be treated as opaque by clients. Only equality comparison on the value is valid.
```

/assign @roycaihw @liggitt @lavalamp

### CODE REVIEW

File: test/cmd/discovery.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ run_assert_short_name_tests() {
   output_message=$(kubectl get --raw=/api/v1)
 
   ## test if a short name is exported during discovery
-  kube::test::if_has_string "${output_message}" '{"name":"configmaps","singularName":"","namespaced":true,"kind":"ConfigMap","verbs":\["create","delete","deletecollection","get","list","patch","update","watch"\],"shortNames":\["cm"\]}'
+  kube::test::if_has_string "${output_message}" '{"name":"configmaps","singularName":"","namespaced":true,"kind":"ConfigMap","verbs":\["create","delete","deletecollection","get","list","patch","update","watch"\],"shortNames":\["cm"\]'
```
Comment (caesarxuchao) [inline test/cmd/discovery.sh]: This change is needed because the `output_message` does not end at "shortNames", it also has "storageVersionHash" now.

**Comment by liggitt** [comment]
@sttts and I were discussing which aspects of a CRD should factor into its storage version hash. GIven that pruning and defaulting can change what persists in storage, should the schema of a CRD factor into its storage version hash?

**Comment by caesarxuchao** [comment]
@sttts @liggitt 

I don't think storageVersionHash should include the CRD's OpenAPI schema. In case of changing CRD schema in a backwards compatible way, it's similar to adding a new field with default value to a built-in v1 API, which does not require storageVersionHash change, nor an immediate storage migration.

Or maybe I missed your point?

File: test/cmd/discovery.sh (hunk 1 above)
Comment (lavalamp) [reply test/cmd/discovery.sh]: Please add this in a comment, or add a `,"storageVersionHash":`?

**Comment by lavalamp** [comment]
re: Jordan's question: the issue is that you can do things to CRD schemas that you (hopefully) cannot to built-in resources. E.g. if you remove a field, then there are a bunch of stored items which need to be re-pruned. Technically this should trigger a migration, however it is also (severe) user error. I am OK not triggering the migrator for this condition--we can always add this in the future.

File: test/cmd/discovery.sh (hunk 1 above)
Comment (caesarxuchao) [reply test/cmd/discovery.sh]: Added a `,"storageVersionHash":`

**Comment by caesarxuchao** [comment]
@liggitt PTAL. It'd be great if we can get this in 1.15. Thank you.

**Pull request merged**